### PR TITLE
GEM Validation update

### PR DESCRIPTION
**Pull request by yeckang**
#### PR description:

* Change the directory path for the GEM Validation plots
* Make a consistent naming scheme for GEM Validation plots with PR #34782
* Use DQM plots with `modeRelval` option for release validation
* Remove the duplicate plots of the GEM Validation and DQM
* Small bugfix for plot naming

#### PR validation:
* The branch has tested with the following commands to check if it works in 2 different geometry versions.
```
runTheMatrix.py -w upgrade -l 11611.0
runTheMatrix.py -w upgrade -l 35811.0
```
* The code format has applied with `scram-build code-format`

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by yeckang** [comment]
What I missed is the PR #34560.
We don't do packing-unpacking for phase 2 scenarios after that PR.
For this reason, the `GEMDigiSource` module couldn't find the unpacked digi.
So I fixed the script to use sim digi when we simulate phase 2 scenarios.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc [hunk 1]
```diff
@@ -286,7 +294,8 @@ void GEMStripDigiValidation::analyze(const edm::Event& event, const edm::EventSe
         Int_t pid = simhit.particleType();
         Int_t pid_idx = getPidIdx(pid);
 
-        me_occ_pid_[key3]->Fill(pid_idx);
+        me_occ_pid_[key2]->Fill(pid_idx);
+        me_occ_pid_eta_[key3]->Fill(pid_idx);
```
Comment (jfernan2) [inline Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc:298]: is the same int quantity filled twice?

**Comment by jfernan2** [comment]
The PR description says: Remove the duplicate plots of the GEM Validation and DQM
However, this PR is adding more plots than those removed:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_1_X_2021-08-11-2300+488006/44729/dqm-histo-comparison-summary.html
@yeckang can you please confirm?
I understand this PR is taking into account also https://github.com/cms-sw/cmssw/pull/34614/

**Comment by yeckang** [comment]
@jfernan2
I've removed several plots from the GEM Validation modules. But at the same time, the parts of DQM modules will make plots for release validation.
So what we see from the release validation is coming from GEM validation + DQM now.

File: Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc (hunk 1 above)
Comment (yeckang) [reply Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc:298]: Yes. But they are filled in different histograms.

Comment (yeckang) [reply Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc:298]: Yes.
That quantity will be filled twice in different histogram.
But... the key for the second histogram is wrong.
Let me check the histograms. And I will push it again.

**Comment by civanch** [comment]
please test

File: Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc [hunk 2]
```diff
@@ -82,21 +84,28 @@ void GEMStripDigiValidation::bookHistograms(DQMStore::IBooker& booker,
...
 
       const auto& superChamberVec = station->superChambers();
       if (superChamberVec.empty() || superChamberVec[0] == nullptr) {
         edm::LogError(kLogCategory_) << "Super chambers missing or null for region = " << region_id
                                      << " and station = " << station_id;
       } else {
+        for (const auto& etaPart : superChamberVec[0]->chambers()[0]->etaPartitions()) {
+          Int_t ieta = etaPart->id().ieta();
+          ME3IdsKey key{region_id, station_id, ieta};
+          me_occ_pid_eta_[key] = bookPIDHist(booker, key2, ieta, "sim_occ_pid", "Particle population");
```
Comment (jfernan2) [inline Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc:107]: I am somehow confused by key*, shouldn't this booker have key as argument onstead of key2?

Comment (yeckang) [reply Validation/MuonGEMDigis/plugins/GEMStripDigiValidation.cc:107]: Usually we don't skip the layer part. But several plots are need to skip them.
In this reason, I made a special function for it to keep original form of the `bookPIDHist`.

**Comment by jfernan2** [comment]
+1

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**